addons/udes_stock/models/stock_location.py

```diff
@@ -106,8 +104,90 @@ def _domain_height_category(self):
 
                 If not set then the location will use the format
                 of the nearest ancestor that has a format specified.
-                """
+                """,
+    )
+    u_location_is_countable = fields.Selection(
+        selection=[
+            ("yes", "Yes"),
+            ("no", "No"),
+        ],
+        string="Location Is Countable",
+        help="""
+    Specifies whether the location is countable. If blank, the value of the parent
+    location is used, if applicable.
+    """,
+    )
+    u_is_countable = fields.Boolean(
+        compute="_compute_countable_locations",
+        store=True,
+        string="Is Countable",
+        help="""
+    Computed countable setting to use for this location.
+    The 'Location Is Countable' value set directly on the location will be used if applicable.
+    Otherwise the computed value set on the parent location will be used, if applicable.
+    """,
     )
+    u_countable_state = fields.Selection(
+        selection=[
+            ("empty", "Empty"),
+            ("has_stock", "Has Stock"),
+            ("archived", "Archived"),
+        ],
+        compute="_compute_state",
+        store=True,
+        string="State",
+        help="""
+        Computed field describing the state of a stockable location that is countable, 
+        see u_countable field.""",
+    )
+
+    @api.depends("u_location_is_countable", "location_id", "location_id.u_is_countable")
+    def _compute_countable_locations(self):
```

Comment:
Very minor: Could we rename the computed methods to `_compute_is_countable` and `_compute_countable_state` to fit in the more with the Odoo standard?

Reply:
Ah good point - the name no longer matches really. Will change